pkg/operation/botanist/kubeapiserverexposure.go

```diff
@@ -79,7 +79,11 @@ func (b *Botanist) DeployKubeAPIServerSNI(ctx context.Context) error {
 }
 
 func (b *Botanist) setAPIServerServiceClusterIP(clusterIP string) {
-	b.APIServerClusterIP = clusterIP
+	if len(b.Shoot.Networks.Services.IP)*8 == 32 {
```

Comment:
I do not think that this is the canonical way to identify if the IP is an IPv4 address in go. Not sure what is the best approach, though.

```
if len(b.Shoot.Networks.Services.IP) == net.IPv4len {
```

would probably work, but I would rather prefer

```
if b.Shoot.Networks.Services.IP.To4() != nil {
```

Reply:
`if b.Shoot.Networks.Services.IP.To4() != nil {` should work.